ui/hooks/usePermissionDescriptions.js

```diff
@@ -0,0 +1,98 @@
+import { useMemo } from 'react';
+import {
+  RestrictedMethods,
+  ///: BEGIN:ONLY_INCLUDE_IN(flask)
+  EndowmentPermissions,
+  PermissionNamespaces,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../shared/constants/permissions';
+///: BEGIN:ONLY_INCLUDE_IN(flask)
+import { coinTypeToProtocolName } from '../helpers/utils/util';
+///: END:ONLY_INCLUDE_IN
+import { useI18nContext } from './useI18nContext';
+
+const UNKNOWN_PERMISSION = Symbol('unknown');
+
+/**
+ * @typedef {Object} PermissionLabelObject
+ * @property {string} label - The text label.
+ * @property {string} leftIcon - The left icon.
+ * @property {string} rightIcon - The right icon.
+ */
+
+/**
+ * @returns {(permissionName:string) => PermissionLabelObject}
+ */
+export const usePermissionDescriptions = () => {
+  const t = useI18nContext();
+
+  return useMemo(() => {
```

Comment:
@Gudahtt 

I think I see. "`t` never changes" because it is the `useMemo` function itself. It's the return value from calling the function that may change, not `t` itself. 
`t` is currently:
```
  const t = useMemo(() => {
    return (key, ...args) =>
      getMessage(currentLocale, current, key, ...args) ||
      getMessage(currentLocale, en, key, ...args);
  }, [currentLocale, current, en]);
```
So let's say this is now a regular function in which we pass `t` and the permission name. If `currentLocale` changes, would these translations here update? or should we keep the hook to watch for `currentLocale`?

Reply:
Sorry you're right, `t` does change when the locale changes. I had forgotten that.

> If `currentLocale` changes, would these translations here update? or should we keep the hook to watch for `currentLocale`?

Maybe best explained by example. Right now `ConnectedAccountsPermissions` looks like this:

```javascript
const ConnectedAccountsPermissions = ({ permissions }) => {
  const t = useI18nContext();
  ...
  const getPermissionDescription = usePermissionDescriptions();

  return (
     ...
      <label htmlFor={permissionName}>
        {getPermissionDescription(permissionName).label}
      </label>
    ...
  );
};
```

If we were to create a regular function called `getPermissionDescription` to replace this hook, it could instead look like this:

```javascript
const ConnectedAccountsPermissions = ({ permissions }) => {
  const t = useI18nContext();
  ...

  return (
     ...
      <label htmlFor={permissionName}>
        {getPermissionDescription(t, permissionName).label}
      </label>
    ...
  );
};
```

`getPermissionDescription` is being called each render with the `t` obtained from the `useI18nContext` hook. That hook ensures that the `t` function always uses the correct locale.

Reply:
Thanks for the explanations and suggestions @Gudahtt! All looks good to me.

I'm creating a new PR now. Tiny concern - I noticed you suggested using singular `description` instead of `descriptions`... @rekmarks, and I, spoke about using `descriptions` vs. `description`. I think we were kinda in the middle about this, but we ended up using the plural version to represent multiple descriptions being returned. I'm not sure if you feel strongly about either-or, but please let me know if you do

Reply:
@Gudahtt @rekmarks 

on second thought... the function is a bit chunky so it might be nice to keep the memoization despite `t` as a dependency is negligible with `useMemo` here. any thoughts on this?

Reply:
Haha, define "chunky"?

Reply:
Lolll just a bit of chunk 🤣

Reply:
For the record, the good kind of chunk though this is all very subjective. haha. On my machine, calling `getPermissionDescription(permissionName)` is really just about 0.025 ms on its first call. When memoized with `useMemo`, the time it takes reduces to about 0.005 ms, so we could save about 0.02 ms, or if there are 5 permissions in the list, 0.1 ms each time we rerender this list. Here, this method is called each time we click "Connect" and each time we expand or collapse the "Permissions" list. I guess this could add up to impact performance, but is likely negligible, by itself, to the user. I'm up for keeping it as a hook. After knowing this info, if we think changing this into util function will add simplicity and more value, I'll be happy to make the change too. 

initial call:
<img width="410" alt="Screenshot 2022-03-04 at 10 40 15 PM" src="https://user-images.githubusercontent.com/20778143/156862535-808b18fc-d2b6-4db1-8557-5eb6828f0eb8.png">

initial call and memoized calls:
<img width="381" alt="Screenshot 2022-03-04 at 10 37 30 PM" src="https://user-images.githubusercontent.com/20778143/156862617-41cbab46-8377-41f4-8b95-47bca898b9ad.png">

Reply:
>we ended up using the plural version to represent multiple descriptions being returned

In that example code snippet, a single description is being returned.

>the function is a bit chunky so it might be nice to keep the memoization

The function is basically a switch statement and an object literal. I'm not sure that memoization is helping a whole lot. Cant draw any firm conclusions from those timings you posted because any function gets faster as it is executed repeatedly, because both SpiderMonkey and V8 optimize at runtime (JIT compilation). And even if we do want to memoize it, that can be done more efficiently using Lodash's `memoize` utility function. Basically it does the same thing, except it saves having to construct a new function definition and throw it away every render, which is what's happening with the `useMemo` hook.